plugin/web/extensions/sidebar/core/components/content/common/FieldComponent/Fields/general/Template.tsx

```diff
@@ -22,6 +22,7 @@ const Template: React.FC<BaseFieldProps<"template">> = ({
   onSceneUpdate,
 }) => {
   const [fieldComponents, setFieldComponents] = useState<FieldComponentType[] | undefined>();
+  const latestFieldComponents = useRef<FieldComponentType[] | undefined>();
```

Comment:
If you are adding this, do we still need the fieldComponents useState? Seems like they are going to be synced now.
I am curious though how this useRef helps/fixes anything. Can you explain here or on Slack how this change is helpful?